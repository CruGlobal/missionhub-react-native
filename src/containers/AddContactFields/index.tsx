/* eslint complexity: 0, max-lines: 0, max-lines-per-function: 0 */

import React, { useState, useRef } from 'react';
import { KeyboardAvoidingView, View, TextInput } from 'react-native';
import { useSelector, useDispatch } from 'react-redux';
import { useTranslation } from 'react-i18next';
import { ThunkAction } from 'redux-thunk';
import { AnyAction } from 'redux';

import { orgPermissionSelector } from '../../selectors/people';
import { Flex, Text, Input } from '../../components/common';
import PopupMenu from '../../components/PopupMenu';
import theme from '../../theme';
import { hasOrgPermissions } from '../../utils/common';
import { PersonType } from '../AddContactScreen';
import { useIsMe } from '../../utils/hooks/useIsMe';
import { Organization } from '../../reducers/organizations';
import { RelationshipTypeEnum } from '../../../__generated__/globalTypes';
import { relationshipTypeList } from '../PersonCategoryScreen';
import DropdownIcon from '../../../assets/images/dropdownIcon.svg';
import Avatar from '../../components/Avatar';
import ImagePicker from '../../components/ImagePicker';
import PencilIcon from '../../../assets/images/pencilIcon.svg';
import { RootState } from '../../reducers';

import styles from './styles';

export interface AddContactFieldsProps {
  person: PersonType;
  organization?: Organization;
  onUpdateData: (data: PersonType) => void;
  next: (props: {
    navigateToStageSelection: boolean;
    person: PersonType;
    updatePerson: (person: PersonType) => void;
  }) => ThunkAction<unknown, RootState, {}, AnyAction>;
  testID?: string;
}

const AddContactFields = ({
  person,
  organization,
  onUpdateData,
  next,
}: AddContactFieldsProps) => {
  const { t } = useTranslation('addContact');
  const [currentInputField, changeCurrentInputField] = useState('');
  const dispatch = useDispatch();
  const isMe = useIsMe(person.id);
  const isEdit = !!person.id;
  const personOrgPermission = useSelector(() =>
    orgPermissionSelector({}, { person, organization }),
  );

  const updateField = (field: string, data: string) => {
    switch (field) {
      case 'firstName':
        onUpdateData({ ...person, firstName: data });
        break;
      case 'lastName':
        onUpdateData({ ...person, lastName: data });
        break;
      case 'relationshipType':
        onUpdateData({
          ...person,
          relationshipType: data as RelationshipTypeEnum,
        });
        break;
    }
  };

  const handleImageChange = (data: { data: string }) => {
    onUpdateData({ ...person, picture: data.data });
  };

  const categoryOptions = relationshipTypeList.map(type => {
    return {
      text: t(`categories.${type}`),
      onPress: () => {
        updateField('relationshipType', type);
      },
    };
  });

  const handleStageSelect = () => {
    dispatch(
      next({
        navigateToStageSelection: true,
        person,
        updatePerson: onUpdateData,
      }),
    );
  };

  const firstNameRef = useRef<TextInput>(null);
  const lastNameRef = useRef<TextInput>(null);
  const lastNameFocus = () => lastNameRef.current?.focus();
  // Check if current field is the one the user is focused on in order to change style
  const isCurrentField = (field: string) => currentInputField === field;
  // Disable the name fields if this person has org permission because you are not allowed to edit the names of other mission hub users
  const personHasOrgPermission = hasOrgPermissions(personOrgPermission);
  return (
    <KeyboardAvoidingView style={styles.fieldsWrap} behavior="position">
      <Flex direction="column">
        {isEdit ? (
          <Flex align="center">
            {isMe ? (
              <ImagePicker
                // @ts-ignore
                testID="ImagePicker"
                onSelectImage={handleImageChange}
                circleOverlay={true}
              >
                <Avatar person={person} size="large" />
                <View style={styles.changeAvatarButton}>
                  <PencilIcon color={theme.white} />
                </View>
              </ImagePicker>
            ) : null}
          </Flex>
        ) : (
          <Flex value={2} justify="end" align="center">
            {isMe ? null : (
              <View style={styles.textWrap}>
                <Text style={styles.addPersonText}>{t('prompt')}</Text>
              </View>
            )}
          </Flex>
        )}

        {person.firstName || isCurrentField('firstName') ? (
          <Text style={isEdit ? styles.editLabel : styles.label}>
            {isEdit
              ? t('profileLabels.firstName')
              : t('profileLabels.firstNameNickname')}
          </Text>
        ) : (
          <Text style={isEdit ? styles.editLabel : styles.label}>{}</Text>
        )}

        <Input
          testID="firstNameInput"
          ref={firstNameRef}
          style={{
            color: isEdit ? theme.parakeetBlue : theme.white,
            borderBottomColor: isEdit
              ? theme.extraLightGrey
              : theme.parakeetBlue,
          }}
          editable={!personHasOrgPermission}
          selectionColor={theme.parakeetBlue}
          onChangeText={(firstName: string) =>
            updateField('firstName', firstName)
          }
          value={person.firstName}
          placeholder={
            isCurrentField('firstName')
              ? ''
              : t('profileLabels.firstNameRequired')
          }
          placeholderTextColor={isEdit ? theme.parakeetBlue : theme.white}
          returnKeyType="next"
          blurOnSubmit={false}
          autoFocus={true}
          onFocus={() => changeCurrentInputField('firstName')}
          onBlur={() => changeCurrentInputField('')}
          onSubmitEditing={lastNameFocus}
        />
      </Flex>
      <Flex direction="column">
        {person.lastName || isCurrentField('lastName') ? (
          <Text style={isEdit ? styles.editLabel : styles.label}>
            {t('profileLabels.lastName')}
          </Text>
        ) : (
          <Text style={isEdit ? styles.editLabel : styles.label}>{}</Text>
        )}

        <Input
          testID="lastNameInput"
          ref={lastNameRef}
          style={{
            color: isEdit ? theme.parakeetBlue : theme.white,
            borderBottomColor: isEdit
              ? theme.extraLightGrey
              : theme.parakeetBlue,
          }}
          editable={!personHasOrgPermission}
          selectionColor={theme.parakeetBlue}
          onChangeText={(lastName: string) => updateField('lastName', lastName)}
          value={person.lastName}
          placeholder={
            isCurrentField('lastName')
              ? ''
              : t('profileLabels.lastNameOptional')
          }
          placeholderTextColor={isEdit ? theme.parakeetBlue : theme.white}
          returnKeyType="done"
          blurOnSubmit={false}
          onFocus={() => changeCurrentInputField('lastName')}
          onBlur={() => changeCurrentInputField('')}
        />
      </Flex>
      {isEdit ? (
        <>
          <Flex direction="column">
            <Text style={isEdit ? styles.editLabel : styles.label}>
              {t('stage')}
            </Text>

            <View style={styles.buttonContainer}>
              <Text
                testID="stageSelectButton"
                onPress={() => handleStageSelect()}
                style={styles.categoryText}
              >
                {person.stage?.name || t('stageNull')}
              </Text>
            </View>
          </Flex>

          {!isMe ? (
            <Flex direction="column">
              <Text style={isEdit ? styles.editLabel : styles.label}>
                {t('categoryPrompt')}
              </Text>

              <PopupMenu
                // @ts-ignore
                actions={categoryOptions}
                triggerOnLongPress={false}
              >
                <View style={styles.buttonContainer}>
                  <Text style={styles.categoryText}>
                    {person.relationshipType
                      ? t(`categories.${person.relationshipType}`)
                      : t('categoryNull')}
                  </Text>
                  <DropdownIcon color={theme.lightGrey} />
                </View>
              </PopupMenu>
            </Flex>
          ) : null}
        </>
      ) : null}
    </KeyboardAvoidingView>
  );
};

export default AddContactFields;
