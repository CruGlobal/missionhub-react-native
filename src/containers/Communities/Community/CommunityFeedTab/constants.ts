export const COMMUNITY_FEED = 'nav/COMMUNITY_FEED';
