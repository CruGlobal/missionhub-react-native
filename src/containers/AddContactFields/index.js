import React, { Component, Fragment } from 'react';
import { KeyboardAvoidingView } from 'react-native';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { translate } from 'react-i18next';

import { ORG_PERMISSIONS } from '../../constants';
import { orgPermissionSelector } from '../../selectors/people';
import { Flex, Text, Input, RadioButton } from '../../components/common';
import theme from '../../theme';
import {
  getPersonEmailAddress,
  getPersonPhoneNumber,
} from '../../utils/common';

import styles from './styles';

@translate()
class AddContactFields extends Component {
  state = {
    firstName: '',
    lastName: '',
    email: '',
    phone: '',
    gender: null,
    path: null,
    orgPermission: {},
  };

  componentDidMount() {
    const { person, orgPermission, isJean, organization } = this.props;
    // If person exists, then we are in edit mode
    if (!person) {
      // Default the orgPermission for creating new people to 'Contact'
      if (isJean && organization && organization.id) {
        this.updateField('orgPermission', {
          ...orgPermission,
          permission_id: ORG_PERMISSIONS.CONTACT,
        });
      }
      return;
    }
    const email = getPersonEmailAddress(person) || {};
    const phone = getPersonPhoneNumber(person) || {};

    if (person) {
      const newState = {
        firstName: person.first_name,
        lastName: person.last_name,
        ...(isJean
          ? {
              emailId: email.id,
              email: email.email,
              phoneId: phone.id,
              phone: phone.number,
              gender: person.gender,
              orgPermission,
            }
          : {}),
      };
      this.setState(newState);
      this.props.onUpdateData(newState);
    }
  }

  updateOrgPermission(pId) {
    this.updateField('orgPermission', {
      ...this.state.orgPermission,
      permission_id: pId,
    });
  }

  updateField(field, data) {
    this.setState({ [field]: data }, () => {
      this.props.onUpdateData(this.state);
    });
  }

  render() {
    const {
      t,
      isJean,
      organization,
      myOrgPermissions,
      orgPermission: personOrgPermission,
    } = this.props;
    const {
      firstName,
      lastName,
      email,
      phone,
      gender,
      orgPermission,
    } = this.state;

    // Disable the name fields if this person has org permission because you are not allowed to edit the names of other mission hub users
    const personHasOrgPermission =
      personOrgPermission &&
      (personOrgPermission.permission_id === ORG_PERMISSIONS.USER ||
        personOrgPermission.permission_id === ORG_PERMISSIONS.ADMIN);
    return (
      <KeyboardAvoidingView style={styles.fieldsWrap} behavior="position">
        <Flex direction="column">
          <Text style={styles.label}>
            {t('profileLabels.firstNameRequired')}
          </Text>
          <Input
            ref={c => (this.firstName = c)}
            editable={!personHasOrgPermission}
            onChangeText={t => this.updateField('firstName', t)}
            value={firstName}
            placeholder={t('profileLabels.firstNameRequired')}
            placeholderTextColor={theme.white}
            returnKeyType="next"
            blurOnSubmit={false}
            onSubmitEditing={() => this.lastName.focus()}
          />
        </Flex>
        <Flex direction="column">
          <Text style={styles.label}>{t('profileLabels.lastName')}</Text>
          <Input
            ref={c => (this.lastName = c)}
            editable={!personHasOrgPermission}
            onChangeText={t => this.updateField('lastName', t)}
            value={lastName}
            placeholder={t('profileLabels.lastName')}
            placeholderTextColor={theme.white}
            returnKeyType="next"
            blurOnSubmit={false}
            onSubmitEditing={() => this.email && this.email.focus()}
          />
        </Flex>
        {isJean ? (
          <Fragment>
            <Flex direction="column" key="email">
              <Text style={styles.label}>{t('profileLabels.email')}</Text>
              <Input
                ref={c => (this.email = c)}
                onChangeText={t => this.updateField('email', t)}
                value={email}
                autoCapitalize="none"
                placeholder={t('profileLabels.email')}
                placeholderTextColor={theme.white}
                keyboardType="email-address"
                returnKeyType="next"
                blurOnSubmit={false}
                onSubmitEditing={() => this.phone.focus()}
              />
            </Flex>
            <Flex
              direction="row"
              align="center"
              style={styles.genderRow}
              key="gender"
            >
              <Text style={styles.genderText}>
                {t('profileLabels.gender')}:
              </Text>
              <RadioButton
                style={styles.genderRadioButton}
                onSelect={() => this.updateField('gender', 'Male')}
                checked={gender === 'Male'}
                label={t('gender.male')}
              />
              <RadioButton
                style={styles.genderRadioButton}
                onSelect={() => this.updateField('gender', 'Female')}
                checked={gender === 'Female'}
                label={t('gender.female')}
              />
            </Flex>
            <Flex direction="column" key="phone">
              <Text style={styles.label}>{t('profileLabels.phone')}</Text>
              <Input
                ref={c => (this.phone = c)}
                onChangeText={t => this.updateField('phone', t)}
                value={phone}
                placeholder={t('profileLabels.phone')}
                placeholderTextColor={theme.white}
                keyboardType="phone-pad"
                returnKeyType="done"
                blurOnSubmit={true}
              />
            </Flex>
            {organization && organization.id ? (
              <Fragment>
                <Text style={styles.label}>
                  {t('profileLabels.permissions')}:
                </Text>
                <Flex
                  direction="row"
                  align="center"
                  style={styles.permissionsRow}
                >
                  <RadioButton
                    style={styles.contactRadioButton}
                    onSelect={() =>
                      this.updateOrgPermission(ORG_PERMISSIONS.CONTACT)
                    }
                    checked={
                      orgPermission.permission_id === ORG_PERMISSIONS.CONTACT
                    }
                    label={t('profileLabels.contact')}
                  />
                  {myOrgPermissions &&
                  (myOrgPermissions.permission_id === ORG_PERMISSIONS.USER ||
                    myOrgPermissions.permission_id ===
                      ORG_PERMISSIONS.ADMIN) ? (
                    <RadioButton
                      style={styles.userRadioButton}
                      onSelect={() =>
                        this.updateOrgPermission(ORG_PERMISSIONS.USER)
                      }
                      checked={
                        orgPermission.permission_id === ORG_PERMISSIONS.USER
                      }
                      label={t('profileLabels.user')}
                    />
                  ) : null}
                  {myOrgPermissions &&
                  myOrgPermissions.permission_id === ORG_PERMISSIONS.ADMIN ? (
                    <RadioButton
                      style={styles.adminRadioButton}
                      onSelect={() =>
                        this.updateOrgPermission(ORG_PERMISSIONS.ADMIN)
                      }
                      checked={
                        orgPermission.permission_id === ORG_PERMISSIONS.ADMIN
                      }
                      label={t('profileLabels.admin')}
                    />
                  ) : null}
                </Flex>
              </Fragment>
            ) : null}
          </Fragment>
        ) : null}
      </KeyboardAvoidingView>
    );
  }
}

AddContactFields.propTypes = {
  person: PropTypes.object,
  onUpdateData: PropTypes.func.isRequired,
};

const mapStateToProps = ({ auth }, { person, organization }) => ({
  myOrgPermissions:
    auth &&
    organization &&
    organization.id &&
    orgPermissionSelector(null, {
      person: auth.person,
      organization: { id: organization.id },
    }),
  orgPermission:
    person &&
    organization &&
    organization.id &&
    orgPermissionSelector(null, {
      person,
      organization: { id: organization.id },
    }),
});
export default connect(mapStateToProps)(AddContactFields);
