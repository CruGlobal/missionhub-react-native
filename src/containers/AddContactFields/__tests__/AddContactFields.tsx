/* eslint max-lines: 0 */
import React from 'react';
import { fireEvent } from 'react-native-testing-library';

import { renderWithContext } from '../../../../testUtils';
import { ORG_PERMISSIONS } from '../../../constants';
import { orgPermissionSelector } from '../../../selectors/people';

import AddContactFields from '..';

jest.mock('../../../selectors/people');

const orgPermission = { permission_id: ORG_PERMISSIONS.CONTACT };

const initialState = { auth: { person: {} } };
const onUpdateData = jest.fn();

beforeEach(() => {
  ((orgPermissionSelector as unknown) as jest.Mock).mockReturnValue(
    orgPermission,
  );
});

it('renders correctly | No Person', () => {
  const { snapshot } = renderWithContext(
    <AddContactFields
      organization={null}
      onUpdateData={onUpdateData}
      person={{}}
    />,
    {
      initialState,
    },
  );
  snapshot();
  expect(onUpdateData).toHaveBeenCalledWith({
    firstName: '',
    lastName: '',
  });
});

it('render correctly | With Person', () => {
  const { snapshot } = renderWithContext(
    <AddContactFields
      organization={null}
      onUpdateData={onUpdateData}
      person={{ first_name: 'Christian', last_name: 'Huffman' }}
    />,
    {
      initialState,
    },
  );
  snapshot();
  expect(onUpdateData).toHaveBeenCalledWith({
    firstName: 'Christian',
    lastName: 'Huffman',
  });
});

describe('calls methods', () => {
  it('calls update firstName and changeFocusedField', async () => {
    const { recordSnapshot, diffSnapshot, getByTestId } = renderWithContext(
      <AddContactFields
        organization={null}
        onUpdateData={onUpdateData}
        person={{}}
      />,
      {
        initialState,
      },
    );
    recordSnapshot();
    await fireEvent(getByTestId('firstNameInput'), 'onFocus');
    await fireEvent.changeText(getByTestId('firstNameInput'), 'Christian');
    diffSnapshot();
    expect(onUpdateData).toHaveBeenLastCalledWith({
      firstName: 'Christian',
      lastName: '',
    });
  });

  it('calls update lastName and changeFocusedField', async () => {
    const { recordSnapshot, diffSnapshot, getByTestId } = renderWithContext(
      <AddContactFields
        organization={null}
        onUpdateData={onUpdateData}
        person={{}}
      />,
      {
        initialState,
      },
    );
    recordSnapshot();
    await fireEvent(getByTestId('lastNameInput'), 'onFocus');
    await fireEvent.changeText(getByTestId('lastNameInput'), 'Huffman');
    diffSnapshot();
    expect(onUpdateData).toHaveBeenLastCalledWith({
      firstName: '',
      lastName: 'Huffman',
    });
  });
});
