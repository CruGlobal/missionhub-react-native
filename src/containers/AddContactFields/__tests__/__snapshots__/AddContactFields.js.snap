// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders casey view correctly 1`] = `
<KeyboardAvoidingView
  behavior="position"
  enabled={true}
  keyboardVerticalOffset={0}
  style={
    Object {
      "paddingHorizontal": 25,
      "paddingVertical": 50,
    }
  }
>
  <Flex
    direction="column"
  >
    <MyText
      style={
        Object {
          "color": "#52C5DC",
          "fontSize": 13,
          "marginTop": 15,
        }
      }
    >
      First Name (Required)
    </MyText>
    <Input
      blurOnSubmit={false}
      editable={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="First Name (Required)"
      placeholderTextColor="#ffffff"
      returnKeyType="next"
    />
  </Flex>
  <Flex
    direction="column"
  >
    <MyText
      style={
        Object {
          "color": "#52C5DC",
          "fontSize": 13,
          "marginTop": 15,
        }
      }
    >
      Last Name
    </MyText>
    <Input
      blurOnSubmit={false}
      editable={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Last Name"
      placeholderTextColor="#ffffff"
      returnKeyType="next"
    />
  </Flex>
</KeyboardAvoidingView>
`;

exports[`renders jean invite with organization and user and admin radio buttons 1`] = `
<KeyboardAvoidingView
  behavior="position"
  enabled={true}
  keyboardVerticalOffset={0}
  style={
    Object {
      "paddingHorizontal": 25,
      "paddingVertical": 50,
    }
  }
>
  <Flex
    direction="column"
  >
    <MyText
      style={
        Object {
          "color": "#52C5DC",
          "fontSize": 13,
          "marginTop": 15,
        }
      }
    >
      First Name (Required)
    </MyText>
    <Input
      blurOnSubmit={false}
      editable={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="First Name (Required)"
      placeholderTextColor="#ffffff"
      returnKeyType="next"
    />
  </Flex>
  <Flex
    direction="column"
  >
    <MyText
      style={
        Object {
          "color": "#52C5DC",
          "fontSize": 13,
          "marginTop": 15,
        }
      }
    >
      Last Name
    </MyText>
    <Input
      blurOnSubmit={false}
      editable={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Last Name"
      placeholderTextColor="#ffffff"
      returnKeyType="next"
    />
  </Flex>
  <Flex
    direction="column"
    key="email"
  >
    <MyText
      style={
        Object {
          "color": "#52C5DC",
          "fontSize": 13,
          "marginTop": 15,
        }
      }
    >
      Email
    </MyText>
    <Input
      autoCapitalize="none"
      blurOnSubmit={false}
      keyboardType="email-address"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Email"
      placeholderTextColor="#ffffff"
      returnKeyType="next"
    />
  </Flex>
  <MyText
    style={
      Object {
        "color": "#52C5DC",
        "fontSize": 13,
        "marginTop": 15,
      }
    }
  >
    Permissions
    :
  </MyText>
  <Flex
    align="center"
    direction="row"
    style={
      Object {
        "paddingVertical": 15,
      }
    }
  />
</KeyboardAvoidingView>
`;

exports[`renders jean with organization and user and admin radio buttons 1`] = `
<KeyboardAvoidingView
  behavior="position"
  enabled={true}
  keyboardVerticalOffset={0}
  style={
    Object {
      "paddingHorizontal": 25,
      "paddingVertical": 50,
    }
  }
>
  <Flex
    direction="column"
  >
    <MyText
      style={
        Object {
          "color": "#52C5DC",
          "fontSize": 13,
          "marginTop": 15,
        }
      }
    >
      First Name (Required)
    </MyText>
    <Input
      blurOnSubmit={false}
      editable={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="First Name (Required)"
      placeholderTextColor="#ffffff"
      returnKeyType="next"
    />
  </Flex>
  <Flex
    direction="column"
  >
    <MyText
      style={
        Object {
          "color": "#52C5DC",
          "fontSize": 13,
          "marginTop": 15,
        }
      }
    >
      Last Name
    </MyText>
    <Input
      blurOnSubmit={false}
      editable={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Last Name"
      placeholderTextColor="#ffffff"
      returnKeyType="next"
    />
  </Flex>
  <Flex
    direction="column"
    key="email"
  >
    <MyText
      style={
        Object {
          "color": "#52C5DC",
          "fontSize": 13,
          "marginTop": 15,
        }
      }
    >
      Email
    </MyText>
    <Input
      autoCapitalize="none"
      blurOnSubmit={false}
      keyboardType="email-address"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Email"
      placeholderTextColor="#ffffff"
      returnKeyType="next"
    />
  </Flex>
  <Flex
    align="center"
    direction="row"
    key="gender"
    style={
      Object {
        "borderBottomColor": "#52C5DC",
        "borderBottomWidth": 1,
        "paddingVertical": 15,
      }
    }
  >
    <MyText
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 16,
        }
      }
    >
      Gender
      :
    </MyText>
    <RadioButton
      checked={false}
      label="Male"
      onSelect={[Function]}
      style={
        Object {
          "paddingHorizontal": 15,
        }
      }
    />
    <RadioButton
      checked={false}
      label="Female"
      onSelect={[Function]}
      style={
        Object {
          "paddingHorizontal": 15,
        }
      }
    />
  </Flex>
  <Flex
    direction="column"
    key="phone"
  >
    <MyText
      style={
        Object {
          "color": "#52C5DC",
          "fontSize": 13,
          "marginTop": 15,
        }
      }
    >
      Phone
    </MyText>
    <Input
      blurOnSubmit={true}
      keyboardType="phone-pad"
      onChangeText={[Function]}
      placeholder="Phone"
      placeholderTextColor="#ffffff"
      returnKeyType="done"
    />
  </Flex>
  <MyText
    style={
      Object {
        "color": "#52C5DC",
        "fontSize": 13,
        "marginTop": 15,
      }
    }
  >
    Permissions
    :
  </MyText>
  <Flex
    align="center"
    direction="row"
    style={
      Object {
        "paddingVertical": 15,
      }
    }
  >
    <RadioButton
      checked={true}
      label="Contact"
      onSelect={[Function]}
      pressProps={
        Array [
          "2",
        ]
      }
      style={
        Object {
          "paddingRight": 25,
        }
      }
    />
  </Flex>
</KeyboardAvoidingView>
`;

exports[`renders jean with organization and user radio buttons 1`] = `
<KeyboardAvoidingView
  behavior="position"
  enabled={true}
  keyboardVerticalOffset={0}
  style={
    Object {
      "paddingHorizontal": 25,
      "paddingVertical": 50,
    }
  }
>
  <Flex
    direction="column"
  >
    <MyText
      style={
        Object {
          "color": "#52C5DC",
          "fontSize": 13,
          "marginTop": 15,
        }
      }
    >
      First Name (Required)
    </MyText>
    <Input
      blurOnSubmit={false}
      editable={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="First Name (Required)"
      placeholderTextColor="#ffffff"
      returnKeyType="next"
    />
  </Flex>
  <Flex
    direction="column"
  >
    <MyText
      style={
        Object {
          "color": "#52C5DC",
          "fontSize": 13,
          "marginTop": 15,
        }
      }
    >
      Last Name
    </MyText>
    <Input
      blurOnSubmit={false}
      editable={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Last Name"
      placeholderTextColor="#ffffff"
      returnKeyType="next"
    />
  </Flex>
  <Flex
    direction="column"
    key="email"
  >
    <MyText
      style={
        Object {
          "color": "#52C5DC",
          "fontSize": 13,
          "marginTop": 15,
        }
      }
    >
      Email
    </MyText>
    <Input
      autoCapitalize="none"
      blurOnSubmit={false}
      keyboardType="email-address"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Email"
      placeholderTextColor="#ffffff"
      returnKeyType="next"
    />
  </Flex>
  <Flex
    align="center"
    direction="row"
    key="gender"
    style={
      Object {
        "borderBottomColor": "#52C5DC",
        "borderBottomWidth": 1,
        "paddingVertical": 15,
      }
    }
  >
    <MyText
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 16,
        }
      }
    >
      Gender
      :
    </MyText>
    <RadioButton
      checked={false}
      label="Male"
      onSelect={[Function]}
      style={
        Object {
          "paddingHorizontal": 15,
        }
      }
    />
    <RadioButton
      checked={false}
      label="Female"
      onSelect={[Function]}
      style={
        Object {
          "paddingHorizontal": 15,
        }
      }
    />
  </Flex>
  <Flex
    direction="column"
    key="phone"
  >
    <MyText
      style={
        Object {
          "color": "#52C5DC",
          "fontSize": 13,
          "marginTop": 15,
        }
      }
    >
      Phone
    </MyText>
    <Input
      blurOnSubmit={true}
      keyboardType="phone-pad"
      onChangeText={[Function]}
      placeholder="Phone"
      placeholderTextColor="#ffffff"
      returnKeyType="done"
    />
  </Flex>
  <MyText
    style={
      Object {
        "color": "#52C5DC",
        "fontSize": 13,
        "marginTop": 15,
      }
    }
  >
    Permissions
    :
  </MyText>
  <Flex
    align="center"
    direction="row"
    style={
      Object {
        "paddingVertical": 15,
      }
    }
  >
    <RadioButton
      checked={true}
      label="Contact"
      onSelect={[Function]}
      pressProps={
        Array [
          "2",
        ]
      }
      style={
        Object {
          "paddingRight": 25,
        }
      }
    />
  </Flex>
</KeyboardAvoidingView>
`;

exports[`renders jean with organization view correctly 1`] = `
<KeyboardAvoidingView
  behavior="position"
  enabled={true}
  keyboardVerticalOffset={0}
  style={
    Object {
      "paddingHorizontal": 25,
      "paddingVertical": 50,
    }
  }
>
  <Flex
    direction="column"
  >
    <MyText
      style={
        Object {
          "color": "#52C5DC",
          "fontSize": 13,
          "marginTop": 15,
        }
      }
    >
      First Name (Required)
    </MyText>
    <Input
      blurOnSubmit={false}
      editable={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="First Name (Required)"
      placeholderTextColor="#ffffff"
      returnKeyType="next"
    />
  </Flex>
  <Flex
    direction="column"
  >
    <MyText
      style={
        Object {
          "color": "#52C5DC",
          "fontSize": 13,
          "marginTop": 15,
        }
      }
    >
      Last Name
    </MyText>
    <Input
      blurOnSubmit={false}
      editable={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Last Name"
      placeholderTextColor="#ffffff"
      returnKeyType="next"
    />
  </Flex>
  <Flex
    direction="column"
    key="email"
  >
    <MyText
      style={
        Object {
          "color": "#52C5DC",
          "fontSize": 13,
          "marginTop": 15,
        }
      }
    >
      Email
    </MyText>
    <Input
      autoCapitalize="none"
      blurOnSubmit={false}
      keyboardType="email-address"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Email"
      placeholderTextColor="#ffffff"
      returnKeyType="next"
    />
  </Flex>
  <Flex
    align="center"
    direction="row"
    key="gender"
    style={
      Object {
        "borderBottomColor": "#52C5DC",
        "borderBottomWidth": 1,
        "paddingVertical": 15,
      }
    }
  >
    <MyText
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 16,
        }
      }
    >
      Gender
      :
    </MyText>
    <RadioButton
      checked={false}
      label="Male"
      onSelect={[Function]}
      style={
        Object {
          "paddingHorizontal": 15,
        }
      }
    />
    <RadioButton
      checked={false}
      label="Female"
      onSelect={[Function]}
      style={
        Object {
          "paddingHorizontal": 15,
        }
      }
    />
  </Flex>
  <Flex
    direction="column"
    key="phone"
  >
    <MyText
      style={
        Object {
          "color": "#52C5DC",
          "fontSize": 13,
          "marginTop": 15,
        }
      }
    >
      Phone
    </MyText>
    <Input
      blurOnSubmit={true}
      keyboardType="phone-pad"
      onChangeText={[Function]}
      placeholder="Phone"
      placeholderTextColor="#ffffff"
      returnKeyType="done"
    />
  </Flex>
  <MyText
    style={
      Object {
        "color": "#52C5DC",
        "fontSize": 13,
        "marginTop": 15,
      }
    }
  >
    Permissions
    :
  </MyText>
  <Flex
    align="center"
    direction="row"
    style={
      Object {
        "paddingVertical": 15,
      }
    }
  >
    <RadioButton
      checked={true}
      label="Contact"
      onSelect={[Function]}
      pressProps={
        Array [
          "2",
        ]
      }
      style={
        Object {
          "paddingRight": 25,
        }
      }
    />
  </Flex>
</KeyboardAvoidingView>
`;

exports[`renders jean without organization view correctly 1`] = `
<KeyboardAvoidingView
  behavior="position"
  enabled={true}
  keyboardVerticalOffset={0}
  style={
    Object {
      "paddingHorizontal": 25,
      "paddingVertical": 50,
    }
  }
>
  <Flex
    direction="column"
  >
    <MyText
      style={
        Object {
          "color": "#52C5DC",
          "fontSize": 13,
          "marginTop": 15,
        }
      }
    >
      First Name (Required)
    </MyText>
    <Input
      blurOnSubmit={false}
      editable={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="First Name (Required)"
      placeholderTextColor="#ffffff"
      returnKeyType="next"
    />
  </Flex>
  <Flex
    direction="column"
  >
    <MyText
      style={
        Object {
          "color": "#52C5DC",
          "fontSize": 13,
          "marginTop": 15,
        }
      }
    >
      Last Name
    </MyText>
    <Input
      blurOnSubmit={false}
      editable={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Last Name"
      placeholderTextColor="#ffffff"
      returnKeyType="next"
    />
  </Flex>
  <Flex
    direction="column"
    key="email"
  >
    <MyText
      style={
        Object {
          "color": "#52C5DC",
          "fontSize": 13,
          "marginTop": 15,
        }
      }
    >
      Email
    </MyText>
    <Input
      autoCapitalize="none"
      blurOnSubmit={false}
      keyboardType="email-address"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Email"
      placeholderTextColor="#ffffff"
      returnKeyType="next"
    />
  </Flex>
  <Flex
    align="center"
    direction="row"
    key="gender"
    style={
      Object {
        "borderBottomColor": "#52C5DC",
        "borderBottomWidth": 1,
        "paddingVertical": 15,
      }
    }
  >
    <MyText
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 16,
        }
      }
    >
      Gender
      :
    </MyText>
    <RadioButton
      checked={false}
      label="Male"
      onSelect={[Function]}
      style={
        Object {
          "paddingHorizontal": 15,
        }
      }
    />
    <RadioButton
      checked={false}
      label="Female"
      onSelect={[Function]}
      style={
        Object {
          "paddingHorizontal": 15,
        }
      }
    />
  </Flex>
  <Flex
    direction="column"
    key="phone"
  >
    <MyText
      style={
        Object {
          "color": "#52C5DC",
          "fontSize": 13,
          "marginTop": 15,
        }
      }
    >
      Phone
    </MyText>
    <Input
      blurOnSubmit={true}
      keyboardType="phone-pad"
      onChangeText={[Function]}
      placeholder="Phone"
      placeholderTextColor="#ffffff"
      returnKeyType="done"
    />
  </Flex>
</KeyboardAvoidingView>
`;
